pkg/apis/wksprovider/controller/wksctl/machine_actuator.go

```diff
@@ -568,17 +568,42 @@ func (a *MachineActuator) update(ctx context.Context, cluster *clusterv1.Cluster
 	return nil
 }
 
+// Runs upgrade commands based on the operating system
+func getInstallPlan(installer *os.OS, k8sVersion string) map[string]plan.Resource {
```

Comment:
I don't know how important it is, but we've been handling these sorts of nested collections of resources as sub-plans in the past. Rather than create an array of resources and passing them to a new method, we just created another builder, added each resource to the sub-plan and then added the sub-plan as a resource to the outer plan. Though I doubt we'll be "undoing" the upgrade, this format will defeat our undo mechanism unless we add support for it.

Reply:
Hi Jerry. Could you please check the changes and let me know what do you think about them?